### PR TITLE
Clean allure-results folder before test started and generate html rep…

### PR DESCRIPTION
**Pull request by phamthai1912**
Clean allure-results folder before test started and generate html report after tests completed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 node_modules
-/target/**
-allure-results/
+api-test/target/**
```
Comment (mynguyenlgg) [inline .gitignore:2]: Should we use wildcards instead of specific folders?
*/target/**
*/allure-results/
*/allure-report/

File: api-test/pom.xml [hunk 2]
```diff
@@ -34,6 +38,12 @@
                     <argLine>
                         -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
                     </argLine>
+                    <properties>
```
Comment (vantruongt2) [inline api-test/pom.xml:41]: I thought we don't need it

File: api-test/src/main/java/com/client/listeners/TestListener.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        log.info(String.format("Delete all files in %s folder", resultDir));
+        try {
+            FileUtils.deleteDirectory(new File(resultDir));
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    public void onFinish(ITestContext context) {
+        final Configuration configuration = new ConfigurationBuilder().useDefault().build();
```
Comment (vantruongt2) [inline api-test/src/main/java/com/client/listeners/TestListener.java:35]: We should generate the single html file on-demand.

File: .gitignore (hunk 1 above)
Comment (phamthai1912) [reply .gitignore]: Updated

File: api-test/pom.xml (hunk 2 above)
Comment (phamthai1912) [reply api-test/pom.xml]: Removed

File: api-test/src/main/java/com/client/listeners/TestListener.java (hunk 3 above)
Comment (phamthai1912) [reply api-test/src/main/java/com/client/listeners/TestListener.java]: Updated

**Review by mynguyenlgg — APPROVED** [review]

**Pull request merged**